File: src/basic_training/Day14_5.js
function solution(arr, queries) {
	queries.forEach(query => {
		const [startIndex, endIndex] = query;
		for (i = startIndex; i <= endIndex; i++) {
			arr[i]++;
		}
	})
	return arr;
}

// 번외 문제풀이 - for 2중 반복문
// function solution(arr, queries) {
// 	for (let i = 0; i < queries.length; i++) {
// 		let [startIndex, endIndex] = queries[i];
// 		for (let j = startIndex; j <= endIndex; j++) {
// 			arr[j] += 1;
// 		}
// 	}
// 	return arr;
// }

File: src/basic_training/Day14_5.java
package basic_training;

public class Day14_5 {
	public int[] solution(int[] arr, int[][] queries) {
		for(int[] query : queries) {
			int startIdx = query[0];
			int endIdx = query[1];

			for(int i=startIdx; i<=endIdx; i++) {
				arr[i] += 1; 
			}
		}
		return arr;
	}
}

// 번외 문제풀이
// for(int i=0; i<queries.length; i++) {
// 	for(int j=queries[i][0]; j<=queries[i][1]; j++) {
// 		arr[j]++;
// 	}
// }
